pkg/cmd/render/feature_set.go

```diff
@@ -0,0 +1,22 @@
+package render
+
+import (
+	"os"
+	"regexp"
+
+	"github.com/openshift/library-go/pkg/assets"
+)
+
+var re = regexp.MustCompile(`\b(Default|TechPreviewNoUpgrade|CustomNoUpgrade|LatencySensitive)\b`)
+
+// FeatureSetFilePredicate returns a FileInfoPredicate fn that filters assets
+// based on the FeatureSet embedded in the asset file name.
+func FeatureSetFilePredicate(featureSet string) assets.FileInfoPredicate {
+	if len(featureSet) == 0 {
+		featureSet = "Default"
+	}
+	return func(info os.FileInfo) bool {
```

Comment:
the filename is convention, but has no enforcement, you're looking for this annotation containing TechPreviewNoUpgrade (it can contain mutiple) https://github.com/openshift/api/blob/master/config/v1/0000_10_config-operator_01_infrastructure-TechPreviewNoUpgrade.crd.yaml#L9

A skim of the code suggests the original library-go rendering library lacks sufficient information in the predicate function to do this, so it has either be post-processed or the library-go function needs an update

Reply:
If you use library-go `ReadUnstructuredOrDie`, then you can use the resulting unstructured `GetAnnotations` to get the map you need to search.

Reply:
Made updates to library-go in https://github.com/openshift/library-go/pull/1421.